File: Chapter 03/C#/3.3.1 SRP/After/Player.cs
using System;

namespace Chapter03.SRP.After;

// Responsibility 1: Manages only the player’s state and actions
public class Player
{
    public required string Name { get; set; }
    
    public void DribbleBall() 
    {
        Console.WriteLine($"  [Action] {Name} is dribbling the ball down the court.");
    }
}

File: Chapter 03/JS/3.3.1 SRP/After/player.js
// Responsibility 1: Manages only the player’s state and actions
class Player {
    constructor(name) {
        this.name = name;
    }

    dribbleBall() {
        console.log(`  [Action] ${this.name} is dribbling the ball down the court.`);
    }
}

module.exports = Player;